tests/test_read_handling.py

```diff
@@ -777,3 +777,46 @@ def test_extract_paired_reads_5_stdin_error():
     status, out, err = utils.runscript(script, args, fail_ok=True)
     assert status == 1
     assert "output filenames must be provided." in err
+
+
+def test_read_bundler():
+    infile = utils.get_test_data('unclean-reads.fastq')
+    records = [r for r in screed.open(infile)]
+    bundle = khmer.utils.ReadBundle(*records)
+
+    raw_reads = (
+        'GGTTGACGGGGNNNAGGGGGCGGCTGACTCCGAGAGACAGCAGCCGCAGCTGTCGTCAGGGGATTTCCG'
+        'GGGCGGAGGCCGCAGACGCGAGTGGTGGAGG',
+        'GGTTGACGGGGCTCAGGGGGCGGCTGACTCCGAGAGACAGCAGCCGCAGCTGTCGTCAGGGGANNNCCG'
+        'GGGCGGAGGCCGCAGACGCGAGTGGTGGAGG',
+    )
+
+    cleaned_reads = (
+        'GGTTGACGGGGAAAAGGGGGCGGCTGACTCCGAGAGACAGCAGCCGCAGCTGTCGTCAGGGGATTTCCG'
+        'GGGCGGAGGCCGCAGACGCGAGTGGTGGAGG',
+        'GGTTGACGGGGCTCAGGGGGCGGCTGACTCCGAGAGACAGCAGCCGCAGCTGTCGTCAGGGGAAAACCG'
+        'GGGCGGAGGCCGCAGACGCGAGTGGTGGAGG',
+    )
+
+    assert bundle.num_reads == 2
+    assert bundle.agg_length == 200
+    assert bundle.cleaned_reads[0] == cleaned_reads[0]
+    assert bundle.cleaned_reads[1] == cleaned_reads[1]
+
+    for (rd, cln), raw, tstcln in zip(bundle.both(), raw_reads, cleaned_reads):
+        assert rd.sequence == raw
+        assert cln == tstcln
+
+    infile = utils.get_test_data('single-read.fq')
```

Comment:
In general each test should look at one kind of behavior, so that you don't fix one thing and still have the test failing. So, please break this up into 4 functions.